pkg/cmd/repo/setdefault/setdefault.go

```diff
@@ -78,14 +78,6 @@ func NewCmdSetDefault(f *cmdutil.Factory, runF func(*SetDefaultOptions) error) *
 		`),
 		Args: cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if len(args) > 0 {
-				var err error
-				opts.Repo, err = ghrepo.FromFullName(args[0])
-				if err != nil {
-					return err
-				}
-			}
-
```

Comment:
**Does it make sense to move processing `opts.Repo` out of `RunE` and into `setDefaultRun`?**

The `spf13/cobra` command hooks should be focused on translating `cobra` arguments and flags into variables that the application logic cares about, however this is no longer as simple as turning `owner/repo` into `ghrepo.Interface`.  With the changes as-is, we're duplicating logic around remotes and starting to do more application-level behavior outside of the application code.

It might be cleaner and easier to test by changing `opts.Repo` to a string like `opts.RepoOrRemote` and defer checking remotes or parsing this as a repo until later in the logic:

https://github.com/cli/cli/blob/9bdfa8367d3e5686979d985073a5f5a78965e346/pkg/cmd/repo/setdefault/setdefault.go#L172-L182

1. Error handling around remotes happened earlier in the logic
2. Application has remote and resolved remote information
3. `RunE` logic keeps this logic within the testable code